corehq/ex-submodules/casexml/apps/case/xform.py

```diff
@@ -302,10 +345,13 @@ def _validate_indices(case):
                         case.get_id,
                         index.referenced_id,
                     )
+                else:
+                    if referenced_case.owner_id != case.owner_id:
+                        dirtiness_flags.append(DirtinessFlag(index.referenced_id, case.owner_id))
```

Comment:
I'm curious to know why you switched from recording `case.case_id` as the hint to `index.referenced_id`. I think either could work but the choice changes the way you check if a hint is still valid. Storing the indexed case means you need to check reverse indices to validate the hint whereas if you store the indexing case you need to check its indices.

Looking at the way we defined it in the doc I think storing the case (and not the referenced case) is correct:
"A hint is still a good hint if (1) its owner is the target owner and (2) it has a parent (indexed case) whose owner is not the target owner. "